addrmgr/addrmanager.go

```diff
@@ -684,10 +684,11 @@ func (a *AddrManager) NeedMoreAddresses() bool {
 	return a.numAddresses() < needAddressThreshold
 }
 
-// AddressCache returns a randomized subset of all known addresses.
+// AddressCache returns a randomized subset of all addresses known to the
+// address manager with a network address type matching the provided type flags.
 //
 // This function is safe for concurrent access.
-func (a *AddrManager) AddressCache() []*NetAddress {
+func (a *AddrManager) AddressCache(supportedNetAddressType NetAddressTypeFilter) []*NetAddress {
```

Comment:
```suggestion
func (a *AddrManager) AddressCache(filter NetAddressTypeFilter) []*NetAddress {
```

Reply:
Also a breaking change to the public API which needs a major module version bump.  I'll stop calling these out from now on since I mentioned it in my previous review and once the major is bumped, the API can be changed at will.